rfcs/text/0000-matrix-service.md

```diff
@@ -0,0 +1,418 @@
+
+### Auxiliary Endpoints
+
+To improve user experience, for each of the four POST body parameters, a pair of `GET` endpoints are available to provide more information about the parameter and the set of accepted values. For example, `GET /filters` and `GET /filters/<filter>` provide respectively, the full list of acceptable filters and information about a specific filter. For filters that take on categorical (string) values, the latter will return a description of the filter, the set of valid values for the filter, and a distribution of cell counts across the values. Numeric filters, such as `genes_detected`, will return a description and the maximum and minimum values across all cells . An example response from each endpoint illustrates this:
+
+Ex. GET /filters
+
+```
+[
+    "cell_suspension.provenance.document_id",
+    "genes_detected",
+    "total_umis",
+    "barcode",
+    "emptydrops_is_cell",
+    "specimen_from_organism.provenance.document_id",
+    "cell_suspension.genus_species.ontology",
+    "cell_suspension.genus_species.ontology_label",
+    "donor_organism.provenance.document_id",
+    "donor_organism.human_specific.ethnicity.ontology",
+    "donor_organism.human_specific.ethnicity.ontology_label",
+    "donor_organism.diseases.ontology",
+    "donor_organism.diseases.ontology_label",
+    "donor_organism.development_stage.ontology",
+    "donor_organism.development_stage.ontology_label",
+    "donor_organism.sex",
+    "donor_organism.is_living",
+    "derived_organ_ontology",
+    "derived_organ_label",
+    "derived_organ_parts_ontology",
+    "derived_organ_parts_label",
+    "specimen_from_organism.organ.ontology",
+    "specimen_from_organism.organ.ontology_label",
+    "specimen_from_organism.organ_parts.ontology",
+    "specimen_from_organism.organ_parts.ontology_label",
+    "library_preparation_protocol.provenance.document_id",
+    "library_preparation_protocol.input_nucleic_acid_molecule.ontology",
+    "library_preparation_protocol.input_nucleic_acid_molecule.ontology_label",
+    "library_preparation_protocol.library_construction_method.ontology",
+    "library_preparation_protocol.library_construction_method.ontology_label",
+    "library_preparation_protocol.end_bias",
+    "library_preparation_protocol.strand",
+    "project.provenance.document_id",
+    "project.project_core.project_short_name",
+    "project.project_core.project_title",
+    "analysis_protocol.provenance.document_id",
+    "dss_bundle_fqid",
+    "bundle_uuid",
+    "bundle_version",
+    "file_uuid",
+    "file_version",
+    "analysis_protocol.protocol_core.protocol_id",
+    "analysis_working_group_approval_status"
+]
+```
+
+Ex. Categorical filter: GET /filters/project.project_core.project_short_name
+
```

Comment:
Not without requesting the matrix. In your example, I imagine the number of combinations of field values could become unmanageable. In the future, possibly a similar API to `POST /matrix` that retrieves cell counts without generating a matrix could be helpful. I'd be interested in seeing what the community interest for this would be. @mckinsel thoughts?